pkg/adapter/network.go

```diff
@@ -106,42 +98,108 @@ func (r *LocalRuntime) NetworkRemove(cli *cliconfig.NetworkRmValues) error {
 	return nil
 }
 
-// getCNIConfigPathByName finds a CNI network by name and
-// returns its configuration file path
-func getCNIConfigPathByName(name, cniConfigPath string) (string, error) {
-	files, err := libcni.ConfFiles(cniConfigPath, []string{".conflist"})
+// NetworkCreate creates a CNI network
+func (r *LocalRuntime) NetworkCreate(cli *cliconfig.NetworkCreateValues) (string, error) {
+	var (
+		err error
+	)
+
+	isGateway := true
+	ipMasq := true
+	subnet := &cli.Network
+	gateway := cli.Gateway
+	ipRange := cli.IPRange
+	// if range is provided, make sure it is "in" network
+
+	// if no network is provided, figure out network
+	if !cli.IsSet("subnet") {
+		subnet, err = network.GetFreeNetwork()
+	} else {
+		// if network is provided, does it conflict with existing CNI or live networks
+		err = network.ValidateUserNetworkIsAvailable(subnet)
+	}
+	if err != nil {
+		return "", err
+	}
+
+	if gateway == nil {
+		// if no gateway is provided, provide it as first ip of network
+		gateway = network.CalcGatewayIP(subnet)
+	}
+	// if network is provided and if gateway is provided, make sure it is "in" network
+	if cli.IsSet("subnet") && cli.IsSet("gateway") {
+		if isGatewayInNetwork := subnet.Contains(gateway); !isGatewayInNetwork {
+			return "", errors.Errorf("gateway %s is not in valid for subnet %s", gateway.String(), subnet.String())
+		}
+	}
+	if cli.IsSet("internal") {
+		isGateway = false
+		ipMasq = false
+
+	}
+
+	// if a range is given, we need to ensure it is "in" the network range.
+	if cli.IsSet("ip-range") {
+		if !cli.IsSet("subnet") {
+			return "", errors.New("you must define a subnet range to define an ip-range")
+		}
+		firstIP, err := network.FirstIPInSubnet(&cli.IPRange)
+		if err != nil {
+			return "", err
+		}
+		lastIP, err := network.LastIPInSubnet(&cli.IPRange)
+		if err != nil {
+			return "", err
+		}
+		if !subnet.Contains(firstIP) || !subnet.Contains(lastIP) {
+			return "", errors.Errorf("the ip range %s does not fall within the subnet range %s", cli.IPRange.String(), subnet.String())
+		}
+	}
+	bridgeDeviceName, err := network.GetFreeDeviceName()
 	if err != nil {
 		return "", err
 	}
-	for _, confFile := range files {
-		conf, err := libcni.ConfListFromFile(confFile)
+	// If no name is given, we give the name of the bridge device
+	name := bridgeDeviceName
+	if len(cli.InputArgs) > 0 {
+		name = cli.InputArgs[0]
+		netNames, err := network.GetNetworkNamesFromFileSystem()
 		if err != nil {
 			return "", err
 		}
-		if conf.Name == name {
-			return confFile, nil
+		if used := network.StringInSliceContains(name, netNames); used {
```

Comment:
No need to set variable

if network.StringInSliceContains(name, netNames) {
...
}